### ISSUE TITLE
linter: Add support for react/no-unused-prop-types

### ISSUE THREAD

**Issue by will-stone** [oxc-project/oxc#23508]
The [react/no-unused-prop-types](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md) can detect if a TypeScript type includes support for a prop that's not being used in the component. This is particularly useful in making sure consumers aren't fed a lie about the API of the component.

I notice there's [a line in the repo](https://github.com/oxc-project/oxc/blob/38c4b0628bf3ff4bff6971bd8f70e354d792877e/tasks/lint_rules/src/unsupported-rules.json#L123) that states:

> "react/no-unused-prop-types": "PropTypes are ignored in React 19, and TypeScript + the `no-unused-vars` rule already catches unused props in most cases."

However, the `no-unused-vars` rule doesn't detect if a component has a TypeScript prop type that's not utilised.

For example

```tsx
import type { FC } from "react"

type Props = {
  readonly a: string
  readonly b: string
}

const Component: FC<Props> = ({ a }) => {
  return <div>{a}</div>
}

export { Component }
```

the `no-unused-prop-types` rule flags `b` as not being used:

<img width="1702" height="620" alt="Image" src="https://github.com/user-attachments/assets/0babeedc-a7a7-4248-b10c-365e0a30ab96" />
